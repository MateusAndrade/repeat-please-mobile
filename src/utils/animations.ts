import { Animated } from 'react-native';

export const timingAnimation = (
	component: Animated.Value,
	toValue: number,
	duration: number,
) => Animated.timing(component, { toValue, duration, useNativeDriver: true });

export const springAnimation = (
	component: Animated.Value,
	toValue: number,
	friction: number,
) => Animated.spring(component, { toValue, friction, useNativeDriver: true });
