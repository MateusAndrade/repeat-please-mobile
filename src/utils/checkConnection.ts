import NetInfo, { NetInfoStateType } from '@react-native-community/netinfo';

export interface ICheckConnection {
	isConnected: boolean;
	type: NetInfoStateType;
}

export const checkConnection = async (): Promise<ICheckConnection> => {
	try {
		return await NetInfo.fetch();
	} catch (error) {
		throw new Error(error.message);
	}
};
