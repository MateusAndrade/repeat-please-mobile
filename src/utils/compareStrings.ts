const letterPairs = (str: string): string[] => {
	const pairs: string[] = [];
	str
		.substring(1)
		.replace(/[^a-zA-Z1-9]/g, '')
		.split('')
		.reduce((result, letter) => {
			pairs.push(result + letter);
			return letter;
		}, str[0]);
	return pairs;
};

const intersection = (pairs1: string[], pairs2: string[]) => {
	const matchPairs: string[] = [];
	pairs1.reduce((pairsReference, pair) => {
		const index = pairsReference.indexOf(pair);
		index >= 0 && matchPairs.push(pairsReference.splice(index, 1).join(''));
		return pairsReference;
	}, pairs2);
	return matchPairs;
};

export const compareString = (str1: string, str2: string) => {
	const pairs1 = letterPairs(str1.toUpperCase());
	const pairs2 = letterPairs(str2.toUpperCase());
	const union = pairs1.length + pairs2.length;
	return (2 * intersection(pairs1, pairs2).length) / union;
};

export const stringsDifference = (strToCheck: string, reference: string) => {
	const matchStrs: { str: string; matches: boolean }[] = [];
	strToCheck
		.toUpperCase()
		.split('')
		.reduce((strsReference, str) => {
			const index = strsReference.indexOf(str);
			matchStrs.push({
				str: index >= 0 ? strsReference.splice(index, 1).join('') : str,
				matches: index >= 0,
			});
			return strsReference;
		}, reference.toUpperCase().split(''));
	return matchStrs;
};

export const hightScoreWord = (str: string, reference: string) =>
	str
		.toUpperCase()
		.split(' ')
		.reduce(
			(highScore, word) => {
				const score = compareString(word, reference);
				return score >= highScore.score ? { word, score } : highScore;
			},
			{ word: '', score: 0 },
		);
