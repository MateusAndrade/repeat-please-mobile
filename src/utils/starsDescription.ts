export const starsDescription = (stars: number): string => {
	switch (stars) {
		case 0 || 1:
			return 'Bad';
		case 2:
			return 'Good';
		case 3:
			return 'excelent';
		default:
			return 'none';
	}
};
