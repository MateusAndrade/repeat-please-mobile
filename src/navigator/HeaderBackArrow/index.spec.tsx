import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { HeaderBackArrow, IHeaderBackArrow } from './';
import { findByDataTest } from '../../utils/testUtil';

const createTestProps = () => ({
	navigation: {
		navigate: jest.fn(),
	},
});

const setup = (props: IHeaderBackArrow): ShallowWrapper => {
	return shallow(<HeaderBackArrow {...props} />);
};

describe('Header Back Arrow basic tests', () => {
	let props: any;
	beforeEach(() => {
		props = createTestProps();
	});

	it('should render Header Back Arrow component properly', () => {
		const wrapper = setup(props);
		const backArrowComponent = findByDataTest(wrapper, 'backArrow');
		expect(backArrowComponent).toHaveLength(1);
		expect(backArrowComponent).toMatchSnapshot();
	});
});
