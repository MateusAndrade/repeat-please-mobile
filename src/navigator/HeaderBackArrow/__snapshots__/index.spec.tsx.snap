// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Back Arrow basic tests should render Header Back Arrow component properly 1`] = `ShallowWrapper {}`;
